### PR TITLE
NIFI-7565 - Support for Kudu DATE type

### PR DESCRIPTION
**Pull request by grishick**
NIFI-7565 Add support for DATE to Kudu NAR bundle
- update Kudu dependencies to Kudu 1.13.0
- add support for passing java.sql.Date for Kudu DATE columns
- add tests for passing java.sql.Date to DATE columns

more about DATE type support in Kudu:
https://issues.apache.org/jira/browse/KUDU-2632

### CODE REVIEW

File: nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java [hunk 1]
```diff
@@ -352,6 +351,9 @@ protected void buildPartialRow(Schema schema, PartialRow row, Record record, Lis
                         case VARCHAR:
                             row.addVarchar(colIdx, DataTypeUtils.toString(value, recordFieldName));
                             break;
+                        case DATE:
+                            row.addDate(colIdx, DataTypeUtils.toDate(value, () -> DataTypeUtils.getDateFormat("yyyy-MM-dd"), recordFieldName));
```
Comment (granthenke) [inline nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java:355]: Maybe use `DataTypeUtils.getDateFormat(RecordFieldType.DATE.getDefaultFormat());` like is done here: https://github.com/apache/nifi/blob/main/nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java#L109

What if the string is in a different format? Do we expect that a user would transform their data? Do other processors handle this?

Comment (grishick) [reply nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java:355]: Hive3 processor uses `org.apache.nifi.serialization.record.DataType.getFormat()` and Solr processor uses `RecordFieldType.DATE.getDefaultFormat()`

Comment (grishick) [reply nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/AbstractKuduProcessor.java]: I pushed a fix as a separate commit

**Comment by granthenke** [comment]
👍 LGTM

**Review by pvillard31 — APPROVED** [review]

**Pull request closed without merging**